src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -67,10 +67,10 @@
      (handler (assoc request :jruby-instance jruby-instance)))))
 
 (defn get-environment-from-request
-  "Gets the environment from a request."
+  "Gets the environment from the URL or query string of a request."
   [req]
-  (-> req
-      (get-in [:params "environment"])))
+  (or (get-in req [:params :environment])
+      (get-in req [:params "environment"])))
```

Comment:
`stringify-keys just looks at the generic `:params` map, stringifying any keywords found for keys.  I was initially just naiively thinking about instead having it walk the generic `:params` map and choose a keyword representation over a string one when both are present.

I like your idea about merging the params from different sources better, though - stringifying keys for each as you go - with `route-params` winning over `query-params`, winning over `form-params` where there are matching keys.  Maybe we should pull all of this up into some higher level middleware the `params` handling in `wrap-params-for-jruby`?  If we did that, seems like we wouldn't even need the above code because we would have normalized to a single "environment" key being in the map at the point this code is being evaluated?